internal/generator/generator.go

```diff
@@ -0,0 +1,92 @@
+package generator
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/nestoca/joy-generator/internal/gitrepo"
+	"github.com/nestoca/joy/api/v1alpha1"
+	"github.com/nestoca/joy/pkg/catalog"
+)
+
+type Generator struct {
+	repo *gitrepo.GitRepo
+}
+
+type Result struct {
+	// Release holds the release's values loaded from the yaml file in the catalog
+	Release *v1alpha1.Release `json:"release"`
+
+	// Environment holds the environment info where the release will be deployed. The full spec is not loaded to minimize the payload size
+	Environment *Environment `json:"environment"`
+
+	// RenderedValues is a yaml string that is the Release.spec.values rendered with any templated fields
+	RenderedValues string `json:"templatedValues"`
+}
+
+type Environment struct {
+	Name        string `json:"name"`
+	ClusterName string `json:"clusterName"`
+	Namespace   string `json:"namespace"`
+}
+
+func New(repo *gitrepo.GitRepo) *Generator {
+	return &Generator{
+		repo: repo,
+	}
+}
+
+// Run runs the generator and returns a slice of results. Each result contains the release, the environment where it
+// will be deployed and the rendered values string.
+func (r *Generator) Run() ([]*Result, error) {
+	// Make sure we have the latest catalog changes
+	err := r.repo.Pull()
+	if err != nil {
+		return nil, fmt.Errorf("pulling git repo: %w", err)
+	}
+
+	// Load Releases and relevant environment info (cluster name & namespace)
+	joyCatalog, err := catalog.Load(catalog.LoadOpts{
+		Dir:          r.repo.Directory(),
+		LoadEnvs:     true,
+		LoadReleases: true,
+		ResolveRefs:  true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("loading catalog: %w", err)
+	}
+
+	var reconciledReleases []*Result
+	for _, releaseGroup := range joyCatalog.Releases.Items {
```

Comment:
Please, let's try to avoid introducing a new name ("release group") for something that already has a name: "cross releases".
```suggestion
	for _, crossRelease := range joyCatalog.Releases.Items {
```

Reply:
sure